### PR TITLE
Add profile postcode editing

### PR DESCRIPTION
**Pull request by the-kenneth**
# Description

Made changes around the select profile screen, profile list hook and how the edit profile is handled.

Now the patient info is loaded when edit profile is pressed and this can then be edited. I also simplified the profile list hook and moved the navigation out of it. 

Additionally I deleted the PatientProfile type which was a subset of Profile, and I just pass that around now. It was a confusing type.

There is 1 todo in the EditLocation screen, i couldnt figure out how to _easily_ push the content to the bottom of the screen. Any ideas welcome

### CODE REVIEW

File: src/features/multi-profile/edit-profile/EditLocationScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                  formikProps={props}
+                  label={i18n.t('edit-profile.location.label')}
+                  placeholder={i18n.t('placeholder-postcode')}
+                  name="postcode"
+                  inputProps={{ autoCompleteType: 'postal-code' }}
+                  showError
+                />
+                {
+                  // TODO : How to push below to the bottom simply?
+                }
```
Comment (the-kenneth) [inline src/features/multi-profile/edit-profile/EditLocationScreen.tsx:77]: This is the todo mentioned above

File: src/core/user/UserService.ts [hunk 2]
```diff
@@ -333,10 +333,11 @@ export default class UserService extends ApiClientBase implements ICoreService {
       patientName = i18n.t('default-profile-name');
     }
 
-    const profile: PatientProfile = {
+    const profile: Profile = {
+      id: patientState.patientId,
       name: patientName,
-      avatarName: (patient.avatar_name || 'profile1') as AvatarName,
-      isPrimaryPatient: !patient.reported_by_another,
+      avatar_name: patient.avatar_name ? patient.avatar_name : DEFAULT_PROFILE,
```
Comment (snack-0verflow) [inline src/core/user/UserService.ts:339]: `avatar_name: patient.avatar_name ?? DEFAULT_PROFILE` A different syntax, not suggesting changes. just sharing

File: src/features/multi-profile/edit-profile/EditLocationScreen.tsx (hunk 1 above)
Comment (snack-0verflow) [reply src/features/multi-profile/edit-profile/EditLocationScreen.tsx:77]: @the-kenneth Maybe flex box's attributes, `justifycontent` with `space-bewteen`? (Maybe)

Comment (the-kenneth) [reply src/features/multi-profile/edit-profile/EditLocationScreen.tsx:77]: Played around with that, couldn't get it working. Not sue if its to do with <form> being embedded

**Pull request merged**